addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -1356,7 +1365,19 @@ export class Rtc extends Record {
             audioTrack.enabled = !this.selfSession.isMute && this.selfSession.isTalking;
             this.state.audioTrack = audioTrack;
             this.linkVoiceActivationDebounce();
-            await this.network?.updateUpload("audio", this.state.audioTrack);
+            if (this.state.screenAudioTrack) {
+                const audioContext = new AudioContext();
+                const micSource = audioContext.createMediaStreamSource(audioStream);
+                const screenAudioStream = new MediaStream([this.state.screenAudioTrack]);
+                const screenAudioSource = audioContext.createMediaStreamSource(screenAudioStream);
+                const mixedAudioDestination = audioContext.createMediaStreamDestination();
+                micSource.connect(mixedAudioDestination);
+                screenAudioSource.connect(mixedAudioDestination);
+                const mixedAudioStreamTrack = mixedAudioDestination.stream.getAudioTracks()[0];
```

Comment:
This should stored in the state, it will be needed if the network changes,

For example when we upgrade from P2P to SFU, or fallback from SFU to P2P, we call `updateUpload()` on the new network to pass all the tracks that already exist.

Most (if not all) `updateUpload("audio", this.state.audioTrack)` should be replaced by `updateUpload("audio", this.state.mixedAudioStreamTrack)`, we could think of `mixedAudioStreamTrack` as an "output audio track" that sometimes is a mix, sometimes is just `audioTrack` and sometimes even just `screenAudioTrack`